oxenc/span.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+#include "common.h"
+
+namespace oxenc {
+inline namespace types { inline namespace span {
+    template <typename T, size_t N = std::dynamic_extent>
+    requires std::is_const_v<T> && std::equality_comparable<T>
+    using const_span = std::span<T, N>;
+
+    using cspan = const_span<const char>;
+    using uspan = const_span<const unsigned char>;
+    using bspan = const_span<const std::byte>;
+}}  // namespace types::span
+
+template <typename T, typename U = T::value_type>
+concept const_span_like = basic_char<U> && std::same_as<const_span<const U>, T>;
+
+namespace detail {
+template <const_span_like T>
+std::string_view to_sv(const T& x) {
+    return {reinterpret_cast<const char*>(x.data()), x.size()};
+}
+}  // namespace detail
+
+inline namespace operators { inline namespace span {
+    template <typename T, size_t N, const_contiguous_range_t<T> R>
+    inline bool operator==(const std::span<T, N>& lhs, const R& rhs) {
+        return std::ranges::equal(lhs, rhs);
+    }
```

Comment:
I think this should be enforcing std::equality_comparable on T instead of generic `typename`:
```suggestion
    template <std::equality_comparable T, size_t N, const_contiguous_range_t<T> R>
    inline bool operator==(const std::span<T, N>& lhs, const R& rhs) {
        return std::ranges::equal(lhs, rhs);
    }
```